src/Standards/PSR2/Sniffs/Classes/ClassDeclarationSniff.php

```diff
@@ -344,19 +344,32 @@ public function processOpen(File $phpcsFile, $stackPtr)
                         $prev = ($className - 1);
                     }
 
-                    $spaceBefore = strlen($tokens[$prev]['content']);
-                    if ($spaceBefore !== 1) {
+                    $last    = $phpcsFile->findPrevious(T_WHITESPACE, $prev, null, true);
+                    $content = $phpcsFile->getTokensAsString(($last + 1), ($prev - $last));
+                    if ($content !== ' ') {
+                        $found = strlen($content);
+
                         $error = 'Expected 1 space before "%s"; %s found';
                         $data  = [
                             $tokens[$className]['content'],
-                            $spaceBefore,
+                            $found,
                         ];
 
                         $fix = $phpcsFile->addFixableError($error, $className, 'SpaceBeforeName', $data);
                         if ($fix === true) {
-                            $phpcsFile->fixer->replaceToken($prev, ' ');
+                            if ($tokens[$prev]['code'] === T_WHITESPACE) {
+                                $phpcsFile->fixer->beginChangeset();
+                                if ($tokens[($prev - 1)]['code'] === T_WHITESPACE) {
+                                    $phpcsFile->fixer->replaceToken(($prev - 1), ' ');
+                                }
+
+                                $phpcsFile->fixer->replaceToken($prev, ' ');
+                                $phpcsFile->fixer->endChangeset();
+                            } else {
+                                $phpcsFile->fixer->addContent($prev, ' ');
```

Comment:
Not sure this is needed. `extendsClassName` would be a parse error, though maybe for `extends/*comment*/ClassName` ? in which case, it would be good to add that as a unit test.

Reply:
`extendsClassName` is invalid, but `extends\ClassName` is valid. I can add test case.

Reply:
I've added more test cases, it seems to be working fine in case with comment and no space after `extends`.